libfastpm/cosmology.c

```diff
@@ -323,7 +333,21 @@ static ode_soln growth_ode_solve(double a, FastPMCosmology * c)
     return soln;
 }
 
-/* FIXME: Each of these functions will redo the ode! That's long! */
+void fastpm_growth_info_init(FastPMGrowthInfo * growth_info, double a, FastPMCosmology * c) {
+    ode_soln soln = growth_ode_solve(a, c);
+    ode_soln soln_a1 = growth_ode_solve(1, c);
+    // FIXME: you could save a=1 soln at the start of code (perhaps to cosmo obj) and then never compute again.
+
+    growth_info->a = a;
+    growth_info->c = c;
+    growth_info->D1 = soln.y0 / soln_a1.y0;
+    growth_info->f1 = soln.y1 / soln.y0;    /* f = d log D / d log a. Note soln.y1 is d d1 / d log a */
+    growth_info->D2 = soln.y2 / soln_a1.y2;
```

Comment:
I recall at one point we had confusion if this was to divide by a1.y2 or a1.y0. Do you remember how we settled the debate? I think we agreed this has been giving consistent results as other codes, but I don't remember why it shall be like this. If you do, adding a comment is probably a good idea.

Reply:
Yes, I've been reading into LPT and I believe this is correct now. Each order of the growth factor, Di, is normalized such that Di(z=0)=1. I'll add a comment.